pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -506,13 +506,23 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) er
 		return nil
 	}
 
-	if pipelineState.IsDone() && pr.IsDone() {
+	if pipelineState.IsDone() && pipelineState.IsFinalTasksDone() && pr.IsDone() {
```

Comment:
what about having IsDone check the final tasks? feels odd that IsDone would return true when the pipeline isnt done?

Reply:
yes, it does feel odd, was trying to keep them separate but I guess we are still talking about the same pipeline so `isDone` can check final tasks too 😕

Reply:
have merged it in the latest PR #2661